cli/internal/fs/testdata/legacy-env/turbo.json

```diff
@@ -0,0 +1,27 @@
+// mocked test comment
+{
+  "pipeline": {
+    // Only legacy declaration
+    "task1": {
+      "dependsOn": ["$TASK_1_VAR_A"]
+    },
+    // Only new declaration
+    "task2": {
+      "env": ["TASK_2_VAR_A"]
+    },
+    // Same var declared in both
+    "task3": {
+      "dependsOn": ["$TASK_3_VAR_A"],
+      "env": ["TASK_3_VAR_A"]
+    },
+    // Different vars declared in both
+    "task4": {
+      "dependsOn": ["$TASK_4_VAR_A"],
+      "env": ["TASK_4_VAR_B"]
+    },
+    // Dollar sign in new key (invalid)
+    "task5": {
+      "env": ["$TASK_5_VAR_A"]
+    }
```

Comment:
> dependsOn: "$123$" // this can only be a file.

We won't be able to support this until we remove existing behavior, right? Or do you want the existence of the file to indicate that it's not an env var?

Reply:
That's just a note that `$123$` can only ever be a file (try to create it as an env var). I don't know that it's worth this edge case, but I did want to mention how it behaved.

Reply:
added in the edge cases you mentioned except the first two